server/website/website/models.py

```diff
@@ -221,6 +223,34 @@ def delete(self, using=DEFAULT_DB_ALIAS, keep_parents=False):
         super(Session, self).delete(using=DEFAULT_DB_ALIAS, keep_parents=False)
 
 
+class SessionKnobManager(models.Manager):
+    @staticmethod
+    def get_knobs_for_session(session):
+            # Returns a dict of the knob
+        knobs = KnobCatalog.objects.filter(dbms=session.dbms)
```

Comment:
This suggestion goes from line 230 up to 239. If you eventually decide it's worth incorporating then it still needs to be double checked to see if it is a perfect replacement to your original code section.

```python
   knobs = list(KnobCatalog.objects.filter(dbms=session.dbms))
   session_knobs = SessionKnob.objects.filter(session=session, knob__in=knobs)
   for sk in session_knobs:
        i = knobs.index(sk.knob)
        knobs[i].minval = sk.minval 
        knobs[i].maxval = sk.maxval
        knobs[i].tunable = sk.tunable

   return [k for k in knobs.values() if k["tunable"]]